examples/database-email-update/index.js

```diff
@@ -7,119 +7,134 @@
+  const updatedTasks = findUpdatedTasks(currentTasks)
+  console.log(`Found ${updatedTasks.length} updated tasks.`)
+  // For each updated task, update local db and send an email notification.
+  for (const task of updatedTasks) {
+    taskPageIdToStatusMap[task.pageId] = task.status
+    await sendUpdateEmailWithSendgrid(task)
+  }
+}
 
-  // Iterate over the current tasks and compare them to tasks in our local store (tasksInDatabase).
-  for (const [key, value] of Object.entries(currTasksInDatabase)) {
-    const page_id = key
-    const curr_status = value.Status
-    // If this task hasn't been seen before.
-    if (!page_id in tasksInDatabase) {
-      // Add this task to the local store of all tasks.
-      tasksInDatabase[page_id] = {
-        Status: curr_status,
-      }
-    } else {
-      // If the current status is different from the status in the local store.
-      if (curr_status !== tasksInDatabase[page_id].Status) {
-        // Change the local store.
-        tasksInDatabase[page_id] = {
-          Status: curr_status,
-        }
-        // Send an email about this change..
-        const msg = {
-          to: process.env.EMAIL_TO_FIELD,
-          from: process.env.EMAIL_FROM_FIELD,
-          subject: "Notion Task Status Updated",
-          text:
-            "A Notion task's: " +
-            value.Title +
-            " status has been updated to " +
-            curr_status +
-            ".",
-        }
-        sendgridMail
-          .send(msg)
-          .then(() => {
-            console.log("Email Sent")
-          })
-          .catch(error => {
-            console.error(error)
-          })
-        console.log("Status Changed")
-      }
+/**
+ * Gets tasks from the database.
+ *
+ * Returns array of task objects with pageId, status, and title.
+ * Array<{ pageId: string, status: string, title: string }>
+ */
+async function getTasksFromNotionDatabase() {
+  const pages = []
+  let cursor = undefined
+  while (true) {
+    console.log("\nFetching tasks from Notion DB...")
+    const { results, next_cursor } = await notion.databases.query({
+      database_id: databaseId,
+      start_cursor: cursor,
+    })
+    pages.push(...results)
+    if (!next_cursor) {
+      break
     }
+    cursor = next_cursor
   }
-  // Run this method every 5 seconds (5000 milliseconds).
-  setTimeout(main, 5000)
+  console.log(`${pages.length} pages successfully fetched.`)
+  return pages.map(page => {
+    const statusProperty = page.properties["Status"]
+    const status = statusProperty ? statusProperty.select.name : "No Status"
+    const title = page.properties["Name"].title
+      .map(({ plain_text }) => plain_text)
+      .join("")
+    return {
+      pageId: page.id,
+      status,
+      title,
+    }
+  })
 }
 
-function main() {
-  findChangesAndSendEmails().catch(console.error)
+/**
+ * Compares task to most recent version of task stored in local store (tasksInDatabase).
```

Comment:
"local db" earlier, "local store" here - also "tasksInDatabase" is not longer a variable name here